fred2/ignoreordersdlg.cpp

```diff
@@ -199,31 +154,32 @@ void ignore_orders_dlg::OnOK()
 	int i;
 	object *objp;
 
+	CCheckListBox* ignore_orders_list = ((CCheckListBox*)GetDlgItem(IDC_IGNORE_ORDERS_LIST));
+
 	// clear out the orders, then set the bits according to which check boxes are set
 	if ( m_ship >= 0 ) {
-		for ( i = 0; i < m_num_checks_active; i++ ) {
-			if ( check_boxes[i].button->GetCheck() )
-				orders_accepted.insert(check_boxes[i].id);
+		for ( i = 0; i < (int) m_orderList.size(); i++) {
+			if (ignore_orders_list->GetCheck(i) == BST_CHECKED)
```

Comment:
Wouldn't ` != BST_UNCHECKED` be more correct, like you have above?

Reply:
In that path it does not matter. It _should_ be impossible for a checkbox to be indeterminate there.